AmplifyPlugins/API/AWSAPICategoryPluginIntegrationTests/GraphQL/GraphQLWithUserPoolIntegrationTests/AuthDirective/GraphQLAuthDirectiveIntegrationTests+Support.swift

```diff
@@ -0,0 +1,174 @@
+                deleteNoteInvoked.fulfill()
+            case .failed(let error):
+                XCTFail("Got failed, error: \(error)")
+            default:
+                XCTFail("Unexpected event: \(event)")
+            }
+        })
+
+        wait(for: [deleteNoteInvoked], timeout: TestCommonConstants.networkTimeout)
+        return result!
+    }
+
+    func queryNote(byId id: String) -> Result<MutationSyncResult?, GraphQLResponseError<MutationSyncResult?>> {
+        let queryNoteInvoked = expectation(description: "note was queried successfully")
+        var result: Result<MutationSyncResult?, GraphQLResponseError<MutationSyncResult?>>?
+        let request = GraphQLRequest<MutationSyncResult?>.query(modelName: SocialNote.modelName, byId: id)
+        _ = Amplify.API.query(request: request) { event in
+            switch event {
+            case .completed(let data):
+                switch data {
+                case .success(let note):
+                    result = .success(note)
+                case .failure(let error):
+                    result = .failure(error)
+                }
+                queryNoteInvoked.fulfill()
+            case .failed(let error):
+                XCTFail("Got failed, error: \(error)")
+            default:
+                XCTFail("Unexpected event: \(event)")
+            }
+        }
+
+        wait(for: [queryNoteInvoked], timeout: TestCommonConstants.networkTimeout)
+        return result!
+    }
+
+    func syncQuery() -> Result<SyncQueryResult, GraphQLResponseError<SyncQueryResult>> {
+        let syncQueryInvoked = expectation(description: "sync query successfully")
+        var result: Result<SyncQueryResult, GraphQLResponseError<SyncQueryResult>>?
+        let request = GraphQLRequest<SyncQueryResult>.syncQuery(modelType: SocialNote.self, limit: 1)
+        _ = Amplify.API.query(request: request) { event in
+            switch event {
+            case .completed(let data):
+                switch data {
+                case .success(let paginatedNotes):
+                    result = .success(paginatedNotes)
+                case .failure(let error):
+                    result = .failure(error)
+                }
+                syncQueryInvoked.fulfill()
+            case .failed(let error):
+                XCTFail("Got failed, error: \(error)")
+            default:
+                XCTFail("Unexpected event: \(event)")
+            }
+        }
+        wait(for: [syncQueryInvoked], timeout: TestCommonConstants.networkTimeout)
+        return result!
+    }
+
+    func getAppSyncError(_ graphQLResponseError: GraphQLResponseError<MutationSyncResult>) -> AppSyncErrorType? {
+        guard case let .error(errors) = graphQLResponseError,
+            let error = errors.first,
+            let extensions = error.extensions,
+            case let .string(errorTypeValue) = extensions["errorType"] else {
+                XCTFail("Other should not be able to update owner's note")
```

Comment:
This error of others not being able to update an owners note seems super specific...  we either need to update the function name, or make this function more generic.